pkg/kv/kvserver/store_raft.go

```diff
@@ -303,7 +303,9 @@ func (s *Store) HandleRaftUncoalescedRequest(
 	// count them.
 	s.metrics.RaftRcvdMessages[req.Message.Type].Inc(1)
 
-	q, _ := s.raftRecvQueues.LoadOrCreate(req.RangeID)
+	// NB: add a buffer for a few extra messages, to allow heartbeats getting
+	// through even if MsgApp quota is maxed out by the sender.
+	q, _ := s.raftRecvQueues.LoadOrCreate(req.RangeID, s.cfg.RaftMaxInflightMsgs+10)
```

Comment:
Let's add a constant for `10`, with an explanation. Beyond that, LGTM!

Reply:
Done.